tests/vcd_edge_gateway.py

```diff
@@ -29,3 +29,20 @@ def test_001_list_edge_gateways(self):
         vdc = VDC(self.client, resource=v)
         edge_gateways = vdc.list_edge_gateways()
         assert len(edge_gateways) > 0
+
+    def test_002_get_nat_rules(self):
+        logged_in_org = self.client.get_org()
+        org = Org(self.client, resource=logged_in_org)
+        v = org.get_vdc(self.config['vcd']['vdc'])
+        vdc = VDC(self.client, resource=v)
+        nat_rules = vdc.get_nat_rules(self.config['vcd']['edge_gateway'])
+        assert len(nat_rules) > 0
+
+    def test_003_get_nat_rules_edge_gateway_not_existent(self):
+        logged_in_org = self.client.get_org()
+        org = Org(self.client, resource=logged_in_org)
+        v = org.get_vdc(self.config['vcd']['vdc'])
+        vdc = VDC(self.client, resource=v)
+        with self.assertRaisesRegex(Exception, 'Edge Gateway not found \''+self.config['vcd']['edge_gateway']+'_not_existent\'') as exception:
```

Comment:
Ideally we should have a specific exception for edge gateway not found error. Which we would catch in this method to mark this test as passed.
Since, such an exception is not currently available to us, we can run the above code block in a try-catch block, and verify that the caught exception's message matches the expected message.

We don't need to print the actual exception content on terminal.

Reply:
as I understand https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex verifies the text of the exception message.